Installing node-canvas@1.6.10 fails with node v10.1.0 (no member 'ForceSet')

<!----------------------------------- STOP! ----------------------------------->
<!---
Having trouble installing node-canvas? Please make sure you have read
the installation instructions located here before asking for help:
https://github.com/Automattic/node-canvas#installation
Still having problems, found a bug or want a feature? Fill out the form below.
-->

<!--- Provide a general summary of the issue in the Title above -->

Installing node-canvas@1.6.10 fails with node v10.1.0. The output of `node-gyp rebuild` includes an error `no member named 'ForceSet' in 'v8::Object'`. Is node-canvas@1 supposed to work with node v10.x?

## Issue or Feature

Before updating to node v10.x (from v8.x) I had a working installation of node-canvas. I assume downgrading to older version and rebuilding should work (but haven't tried). node-canvas@next installs fine, but seems to have breaking changes that require changes in the application code.

Full output:

```
> canvas@1.6.10 install /Users/tristan/Code/Projekte/ahorn/cj/cj-api/node_modules/canvas
> node-gyp rebuild

  SOLINK_MODULE(target) Release/canvas-postbuild.node
  CXX(target) Release/obj.target/canvas/src/Canvas.o
In file included from ../src/Canvas.cc:7:
In file included from ../src/Canvas.h:22:
In file included from ../../nan/nan.h:192:
../../nan/nan_maybe_43_inl.h:112:15: error: no member named 'ForceSet' in 'v8::Object'
  return obj->ForceSet(isolate->GetCurrentContext(), key, value, attribs);
         ~~~  ^
In file included from ../src/Canvas.cc:7:
In file included from ../src/Canvas.h:22:
../../nan/nan.h:834:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context)
      [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/tristan/.node-gyp/10.1.0/include/node/node.h:171:1: note: 'MakeCallback' has been explicitly marked
      deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/tristan/.node-gyp/10.1.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/Canvas.cc:7:
In file included from ../src/Canvas.h:22:
../../nan/nan.h:849:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context)
      [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/tristan/.node-gyp/10.1.0/include/node/node.h:164:1: note: 'MakeCallback' has been explicitly marked
      deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/tristan/.node-gyp/10.1.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/Canvas.cc:7:
In file included from ../src/Canvas.h:22:
../../nan/nan.h:864:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context)
      [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/tristan/.node-gyp/10.1.0/include/node/node.h:157:1: note: 'MakeCallback' has been explicitly marked
      deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/tristan/.node-gyp/10.1.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/Canvas.cc:7:
In file included from ../src/Canvas.h:22:
../../nan/nan.h:1473:31: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context)
      [-Wdeprecated-declarations]
    return scope.Escape(node::MakeCallback(
                              ^
/Users/tristan/.node-gyp/10.1.0/include/node/node.h:171:1: note: 'MakeCallback' has been explicitly marked
      deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/tristan/.node-gyp/10.1.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
../src/Canvas.cc:76:51: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
  if (info[2]->IsString()) type = !strcmp("pdf", *String::Utf8Value(info[2]))
                                                  ^
/Users/tristan/.node-gyp/10.1.0/include/node/v8.h:2821:5: note: 'Utf8Value' has been explicitly marked
      deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/tristan/.node-gyp/10.1.0/include/node/v8config.h:321:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/Canvas.cc:78:23: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
    : !strcmp("svg", *String::Utf8Value(info[2]))
                      ^
/Users/tristan/.node-gyp/10.1.0/include/node/v8.h:2821:5: note: 'Utf8Value' has been explicitly marked
      deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/tristan/.node-gyp/10.1.0/include/node/v8config.h:321:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
6 warnings and 1 error generated.
make: *** [Release/obj.target/canvas/src/Canvas.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:235:12)
gyp ERR! System Darwin 17.5.0
gyp ERR! command "/usr/local/Cellar/node/10.1.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/tristan/Code/Projekte/ahorn/cj/cj-api/node_modules/canvas
gyp ERR! node -v v10.1.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
npm WARN eslint-config-standard@6.2.1 requires a peer of eslint-plugin-promise@>=3.3.0 but none is installed. You must install peer dependencies yourself.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! canvas@1.6.10 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the canvas@1.6.10 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/tristan/.npm/_logs/2018-05-14T12_44_46_264Z-debug.log
```

## Steps to Reproduce
```
# Update to Node 10.1.0
$ npm rebuild
```

## Your Environment
* Version of node-canvas: 1.6.10:
* Environment: node v10.1.0 on Mac OS 10.13.4

--- a/package.json
+++ b/package.json
@@ -29,7 +29,7 @@
     "test-server": "node test/server.js"
   },
   "dependencies": {
-    "nan": "^2.4.0"
+    "nan": "^2.10.0"
   },
   "devDependencies": {
     "body-parser": "^1.13.3",

